### PR TITLE
Global exception handler

### PR DESCRIPTION
**Pull request by VadimBlagiy**
This is duplicate my branch "Added feature rsql and started to make GlobalExceptionHandler"

### CODE REVIEW

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/exceptions/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler
+    public ResponseEntity<IncorrectData> handlerException (NoSuchException exception) {
```
Comment (akarzhaneuski) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/exceptions/GlobalExceptionHandler.java:13]: Please merge this with ``` ErrorHandlingControllerAdvice ``` in master

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java [hunk 2]
```diff
@@ -57,13 +56,20 @@ public ImageResponseDTO save(ImageCreateDTO imageCreateDTO) {
...
+                .orElseThrow(() -> new NoSuchException("Image with ID = " + id + " not found in Database" ));
 
         return imageMapper.convertEntityToDto(imageDAO.save(imageEntity));
     }
 
 
     @Override
-    public void delete(Long id) {
+    public Boolean delete(Long id) {
+        ImageEntity imageRemove = imageDAO.findById(id)
```
Comment (akarzhaneuski) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java:68]: Possible, but i would replace with 
``` 
if ( !imageDAO.existsById()) {
    throw new NoSuchException('...');
}
```
Same for all places where we want to check if object exists in database

**Review by akarzhaneuski — CHANGES_REQUESTED** [review]

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/exceptions/GlobalExceptionHandler.java (hunk 1 above)
Comment (VadimBlagiy) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/exceptions/GlobalExceptionHandler.java:13]: ok

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java (hunk 2 above)
Comment (VadimBlagiy) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java:68]: Thanks! I will!

File: sandbox-web/build.gradle [hunk 3]
```diff
@@ -22,5 +22,7 @@ dependencies {
 //    testImplementation 'org.springframework.security:spring-security-test'
     implementation 'org.modelmapper:modelmapper:2.4.0'
     implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
+    implementation group: 'cz.jirutka.rsql', name: 'rsql-parser', version: '2.1.0'
```
Comment (akarzhaneuski) [inline sandbox-web/build.gradle:25]: We already have this dependency ``` implementation 'cz.jirutka.rsql:rsql-parser:2.1.0' ```, this line should be removed

**Review by akarzhaneuski — CHANGES_REQUESTED** [review]

File: sandbox-web/build.gradle (hunk 3 above)
Comment (VadimBlagiy) [reply sandbox-web/build.gradle:25]: Ok!

**Review by akarzhaneuski — APPROVED** [review]

**Pull request merged**